### PR TITLE
refactor: leverage compareTo method

### LINKED ISSUE #744: fix: improve code quality
You do not have to compare the dates twice - just use the `compareTo()` method.

Analogy: you have
```
if (a == b || a < b) 
```

I am asking you to write:
```
if (a <= b)
```

_Originally posted by @aababilov in https://github.com/MobilityData/gtfs-validator/pull/694#discussion_r575128412_

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #744

**Summary:**

This PR improves code quality by leveraging `compareTo()` method in rule validator.

**Expected behavior:** 

Same tests should pass.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FeedExpirationDateValidator.java [hunk 1]
```diff
@@ -56,8 +56,9 @@ public void validate(GtfsFeedInfo entity, NoticeContainer noticeContainer) {
                 currentDatePlusSevenDays));
         return;
       }
-      if (entity.feedEndDate().equals(currentDatePlusThirtyDays)
-          || entity.feedEndDate().isBefore(currentDatePlusThirtyDays)) {
+      if (entity.feedEndDate().compareTo(currentDatePlusThirtyDays) <= 0) {
+        // compareTo() returns 0 if both dates are equal, or a negative integer if
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FeedExpirationDateValidator.java:60]: The behavior of compareTo() is well known, so you can remove this comment.

**Review by barbeau — COMMENTED** [review]
LGTM, but the comment can be removed

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FeedExpirationDateValidator.java (hunk 1 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FeedExpirationDateValidator.java]: Done, thanks!

**Review by barbeau — APPROVED** [review]
👍

**Comment by aababilov** [comment]
Thanks!

**Pull request merged**